dataset/bin.cpp

```diff
@@ -211,17 +212,17 @@ DataArray add_metadata(std::tuple<DataArray, Variable> &&proto,
       dims.emplace(coord.dims().inner());
       out_coords[coord.dims().inner()] = copy(coord);
     }
-  for (const auto &[dim, coord] : coords)
+  for (const auto &[_dim, coord] : coords)
```

Comment:
Are we fine with leading underscores in C++?

In my understanding, all usages in this PR are fine because the underscore is always followed by a lowercase letter and in function scope. But there are limitations: https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

Reply:
I don't see a clash but I think you raise a good point. I'll change.